apps/todomvc-remix/app/routes/api.auth.callback.google.tsx

```diff
@@ -0,0 +1,61 @@
+import { LoaderFunction, redirect } from "@remix-run/node";
+import { keelAccessCookie, keelRefreshCookie } from "~/cookies.server";
+
+const clientId = process.env.GOOGLE_CLIENT_ID;
+const clientSecret = process.env.GOOGLE_CLIENT_SECRET;
+
+if (!clientId) {
+  throw new Error("Missing GOOGLE_CLIENT_ID");
+}
+
+if (!clientSecret) {
+  throw new Error("Missing GOOGLE_CLIENT_SECRET");
+}
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const { searchParams } = new URL(request.url);
+  const params = Object.fromEntries(searchParams.entries());
+
+  const data = await getIdTokenWithCode(params.code);
+  const tokens = await exchangeGoogleIdTokenForKeelTokens(data.id_token);
+
+  const headers = new Headers([
+    ["Cache-Control", "must-revalidate, no-store, no-cache"],
+    ["Set-Cookie", await keelRefreshCookie.serialize(tokens.refresh_token)],
+    ["Set-Cookie", await keelAccessCookie.serialize(tokens.access_token)],
+  ]);
+
+  return redirect("/", { headers });
+};
+
+const getIdTokenWithCode = async (code: string) => {
+  const response = await fetch(`https://oauth2.googleapis.com/token`, {
+    method: "POST",
+    headers: { "Content-Type": "application/x-www-form-urlencoded" },
+    body: new URLSearchParams({
+      code,
+      clientId,
+      clientSecret,
+      redirectUri: "http://localhost:3000/api/auth/callback/google",
+      grant_type: "authorization_code",
+    }).toString(),
+  });
+  return response.json();
+};
+
+const exchangeGoogleIdTokenForKeelTokens = async (idToken: string) => {
```

Comment:
This all feels fairly easy and straightforward, we can totally wrap up in some client helpers though to make it even better.

Reply:
Anything to change here?

Reply:
No that was more of a "nice work" comment 😄